### PR TITLE
Tephra 309

### PR DESCRIPTION
**Pull request by stoty**

### CODE REVIEW

File: tephra-hbase-compat-1.4-base/pom.xml [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+                <goal>add-test-source</goal>
+              </goals>
+              <configuration>
+                <sources>
+                  <source>${project.parent.basedir}/src/test/java</source>
+                </sources>
+              </configuration>
+            </execution>
+          </executions>
+        </plugin>
```
Comment (joshelser) [inline tephra-hbase-compat-1.4-base/pom.xml:210]: Shouldn't be necessary -- these are included by default.

Comment (stoty) [reply tephra-hbase-compat-1.4-base/pom.xml:210]: This section copies the common java source files from -common to the version specific sub-sub-projects that have no source files at all. See the <sources> element. The same approach is used in all the -common based compatibility modules that share the java code.

**Comment by stoty** [comment]
> `tephra-examples/tephra-examples-post-1.3/hbase-1.6/pom.xml~` shouldn't be staged?

Of course. Thanks for noticing it.

**Comment by stoty** [comment]
removed backup files, rebased, and re-added shaded artifacts.

File: tephra-hbase-compat-1.4-base/pom.xml (hunk 1 above)
Comment (joshelser) [reply tephra-hbase-compat-1.4-base/pom.xml:210]: OH! I missed that this is `project.PARENT.basedir`. Thanks.

**Pull request closed without merging**